nodeJS/express/express_forms_validation.md

```diff
@@ -0,0 +1,520 @@
+### Introduction
+
+This lesson will cover how to secure your Node.js / Express applications by using the `express-validator` module to perform validation and sanitization. Proper form handling is how we maintain data integrity and prevent security vulnerabilities in our web applications.
+
+### Lesson overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- Understanding HTML form elements and their actions.
+- The form handling process in a client-server application.
+- Using `express-validator` for validation and sanitization.
+- Best practices for form design and data integrity.
+
+### HTML forms overview
+
+First, a brief refresher on the actual HTML.
+
+Let's create a simple HTML form, with a single text field for collecting a full name, and its associated label element.
+
+The HTML structure would look something like this:
+
+```html
+<!-- The action /users/create corresponds to a route you'll define in your Express application -->
+<form action="/users/create" method="POST" >
+  <label for="fullName">Full Name:</label>
+  <input placeholder="John Doe" type="text" name="fullName" id="fullName">
+  <button type="submit">Submit</button>
+</form>
+```
+
+We have the HTML form element itself, with an action pointing to some resource on our server, and a method defined. Notice how the method corresponds to an HTTP verb, typically either `GET` or `POST`.
+
+The input needs a corresponding label element, which is what the end user actually sees on the page.
+
+Then, we have our input. It has an `id` and a `name` attribute, which must match the `for` attribute defined on our label above. The `name` attribute is especially important, as it defines how our input will be identified in the form data sent to our server.
+
+It has a type of `text` which tells the browser what type of widget to display, what values to accept, and is used for JavaScript and CSS selectors.
+
+The `type="submit"` button can be pressed by the user to upload the data entered into all other input elements to the server (in this case there's only one).
+
+The `form` attributes define how to communicate with the server:
+
+- `action`: The resource/URL where data is sent for processing when the form is submitted. If this is not set, or is an empty string, the form will be submitted back to the current page URL
+- `method`: Defines the HTTP method to use (`POST` or `GET`).
+
+`POST` is more secure for sensitive information, and is used when changing or creating data in the server's database. `GET` is for forms that don't modify data, such as search forms, or when you want the form submission to be bookmarkable or shareable via URL.
```

Comment:
> `POST` is more secure for sensitive information

Oh interesting, could you share a reference link or something for this?

Reply:
Maybe `secure` is the wrong choice of word here? The idea is that `POST` is more about "security through obscurity." A `GET` request could expose credentials in the URL as seen here: https://owasp.org/www-community/vulnerabilities/Information_exposure_through_query_strings_in_url

I'm open to other ways of rewording this though

Reply:
Cool, we can reword it like you have above.